modin/conftest.py

```diff
@@ -577,3 +586,151 @@ def pytest_sessionfinish(session, exitstatus):
+        url = f"http://127.0.0.1:555{worker_id}/"
+    return {"client_kwargs": {"endpoint_url": url}}
+
+
+@pytest.fixture(scope="session")
+def s3_base(worker_id):
+    """
+    Fixture for mocking S3 interaction.
+    Sets up moto server in separate process locally
+    Return url for motoserver/moto CI service
+    """
+    # copied from pandas conftest.py
+    with pandas._testing.ensure_safe_environment_variables():
+        # still need access keys for https://github.com/getmoto/moto/issues/1924
+        os.environ.setdefault("AWS_ACCESS_KEY_ID", "foobar_key")
+        os.environ.setdefault("AWS_SECRET_ACCESS_KEY", "foobar_secret")
+        os.environ["AWS_REGION"] = "us-west-2"
+        if ModinGithubCI.get():
+            if (
+                sys.platform == "darwin"
+                or (sys.platform == "win32" or sys.platform == "cygwin")
+                or (
+                    platform.machine() in ("arm64", "aarch64")
+                    or platform.machine().startswith("armv")
+                )
+            ):
+                # pandas comments say:
+                # NOT RUN on Windows/macOS/ARM, only Ubuntu
+                # - subprocess in CI can cause timeouts
+                # - GitHub Actions do not support
+                #   container services for the above OSs
+                pytest.skip(
+                    (
+                        "S3 tests do not have a corresponding service in Windows, macOS "
+                        + "or ARM platforms"
+                    )
+                )
+            else:
+                # assume CI has started moto in docker container:
+                # https://docs.getmoto.org/en/latest/docs/server_mode.html#run-using-docker
+                # It would be nice to start moto on another thread as in the
+                # instructions here:
+                # https://docs.getmoto.org/en/latest/docs/server_mode.html#start-within-python
+                # but that gives 403 forbidden error when we try to create the bucket
+                yield "http://localhost:5000"
+        else:
+            # Launching moto in server mode, i.e., as a separate process
+            # with an S3 endpoint on localhost
+
+            worker_id = "5" if worker_id == "master" else worker_id.lstrip("gw")
+            endpoint_port = f"555{worker_id}"
+            endpoint_uri = f"http://127.0.0.1:{endpoint_port}/"
+
+            # pipe to null to avoid logging in terminal
+            # TODO any way to throw the error from here? e.g. i had an annoying problem
+            # where I didn't have flask-cors and moto just failed .if there's an error
+            # in the popen command and we throw an error within the body of the context
+            # manager, the test just hangs forever.
+            with subprocess.Popen(
+                # try this https://stackoverflow.com/a/72084867/17554722 ?
+                shlex.split(f"moto_server s3 -p {endpoint_port}"),
```

Comment:
Seems like `shlex.split` is safer since it splits using shell logic - so if we in the future change this to something including quotes, etc., this will still be correct. I was initially on board with removing this, but I have had problems in the past with getting the split right when working with Docker or subprocess, so I'm ok with keeping it or replacing it, although I'd lean towards keeping.

Reply:
But splitting would just produce the same array, and, if you need quotes _inside_ the arguments, you'd just put them in using this way, and would have to fiddle with escaping if using shell way... plus `shlex` follows Bash rules, and what if we'd like something Windows-like?..

anyway, that's not something I'd die on 🙃